src/tests/conformance/messages.rs

```diff
@@ -178,15 +177,102 @@ async fn ignores_unsolicited_responses() {
             .await
             .unwrap();
 
-        match filter.read_from_stream(&mut stream).await.unwrap() {
-            Message::Pong(returned_nonce) => assert_eq!(nonce, returned_nonce),
-            msg => panic!("Expected pong: {:?}", msg),
-        }
+        let pong = filter.read_from_stream(&mut stream).await.unwrap();
+        assert_eq!(pong.kind(), MessageKind::Pong);
     }
 
     node.stop().await;
 }
 
+#[tokio::test]
+async fn basic_query_response() {
+    // ZG-CONFORMANCE-010
+    //
+    // The node responds with the correct messages. Message correctness is naively verified through successful encoding/decoding.
+    //
+    // `Ping` expects `Pong`.
+    // `GetAddr` expects `Addr`.
+    // `Mempool` expects `Inv`.
+    // `Getblocks` expects `Inv`.
+    // `GetData(tx_hash)` expects `Tx`.
+    // `GetData(block_hash)` expects `Blocks`.
+    // `GetHeaders` expects `Headers`.
+    //
+    // The test currently fails for zcashd; zebra tbd.
+    //
+    // Current behaviour:
+    //
+    //  zcashd: Ignores the following messages
```

Comment:
Its.. inconsistent. I did initially seed it with blocks, and then `GetBlocks`, and `GetData(block)` work as expected.

Then one has `GetData(tx)` which actually returns `NotFound` despite not having data.

I would actually hope that all of these queries should result in an `{empty}` style response, or `NotFound` for the `GetData`.

Essentially I think the node never sends an empty response except for `GetData(tx)` - because the bitcoin protocol lists it as an example:
```
notfound is a response to a getdata, sent if any requested data items could not be relayed, for example, because the requested transaction was not in the memory pool or relay set. 
```

Reply:
Might be good to have both cases? This would also serve as documentation for the current behaviour in each case and make it easier for ECC/ZF to evaluate what the spec should be.

Reply:
I'll split up the test case into seeded and unseeded